tools/export/exporters.py

```diff
@@ -1,245 +1,220 @@
-            for d in copy(dirs):
-                if d == '.' or d == '..':
-                    dirs.remove(d)
-
-            for file in files:
-                file_path = join(root, file)
-                resources.append(file_path)
-
-        return resources
-
-    def scan_and_copy_resources(self, prj_paths, trg_path, relative=False):
-        # Copy only the file for the required target and toolchain
-        lib_builds = []
-        # Create the configuration object
-        if isinstance(prj_paths, basestring):
-            prj_paths = [prj_paths]
-        config = Config(self.target, prj_paths)
-        for src in ['lib', 'src']:
-            resources = self.__scan_and_copy(join(prj_paths[0], src), trg_path)
-            for path in prj_paths[1:]:
-                resources.add(self.__scan_and_copy(join(path, src), trg_path))
-
-            lib_builds.extend(resources.lib_builds)
-
-            # The repository files
-            #for repo_dir in resources.repo_dirs:
-            #    repo_files = self.__scan_all(repo_dir)
-            #    for path in prj_paths:
-            #        self.toolchain.copy_files(repo_files, trg_path, rel_path=join(path, src))
-
-        # The libraries builds
-        for bld in lib_builds:
-            build_url = open(bld).read().strip()
-            lib_data = self.build_url_resolver(build_url)
-            lib_path = lib_data['path'].rstrip('\\/')
-            self.__scan_and_copy(lib_path, join(trg_path, lib_data['name']))
-
-            # Create .hg dir in mbed build dir so it's ignored when versioning
-            hgdir = join(trg_path, lib_data['name'], '.hg')
-            mkdir(hgdir)
-            fhandle = file(join(hgdir, 'keep.me'), 'a')
-            fhandle.close()
-
-        if not relative:
-            # Final scan of the actual exported resources
-            resources = self.toolchain.scan_resources(trg_path)
-            resources.relative_to(trg_path, self.DOT_IN_RELATIVE_PATH)
-        else:
-            # use the prj_dir (source, not destination)
-            resources = self.toolchain.scan_resources(prj_paths[0])
-            for path in prj_paths[1:]:
-                resources.add(toolchain.scan_resources(path))
-
-        # Loads the resources into the config system which might expand/modify resources based on config data
-        self.resources = config.load_resources(resources)
-
-        if hasattr(self, "MBED_CONFIG_HEADER_SUPPORTED") and self.MBED_CONFIG_HEADER_SUPPORTED :
-            # Add the configuration file to the target directory
-            self.config_header = self.toolchain.MBED_CONFIG_FILE_NAME
-            config.get_config_data_header(join(trg_path, self.config_header))
-            self.config_macros = []
-        else:
-            # And add the configuration macros to the toolchain
-            self.config_macros = config.get_config_data_macros()
+        self.project = Project(self.project_name, [project_data], settings)
```

Comment:
Is ``project_data`` required there? Because this structure does not have any ``common`` or ``tool_specific`` keys. Are those data valid? project_data would either be empty list or provide ``common`` and ``tool_specific`` data, otherwise data are not valid (it currently does not handle this use case at all - no debug output or error thrown). This shows that there's no clear distinction between public progen data (those that are allowed in yaml files), and internal data.